avm/ptn/aca-lza/hosting-environment/tests/e2e/hub-spoke/main.test.bicep

```diff
@@ -0,0 +1,87 @@
+metadata name = 'Using a hub and spoke deployment.'
+metadata description = 'This instance deploys the module including a Hub to peer to.'
+
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+// e.g., for a module 'network/private-endpoint' you could use 'npe' as a prefix and then 'waf' as a suffix for the waf-aligned test
+param serviceShort string = 'hubspoke'
+
+@description('Optional. Test name prefix.')
+param namePrefix string = '#_namePrefix_#'
+
+@description('Optional. The password to leverage for the login.')
+@secure()
+param password string = newGuid()
+
+// ================= //
+// Variables Section //
+// ================= //
+
+// ============ //
+// Dependencies //
+// ============ //
+module hubdeployment 'deploy.hub.bicep' = {
```

Comment:
I may suggest to align this deployment of dependencies with the general approach in AVM, that is passing all resource names in from this file (with a `dep`  prefix and both the `serviceShort` &  `namePrefix` being aprt of it). The reasons are that `dep-` resources are automatically excluded from the PSRule validation (they're just a means to an end afterall) and also that they can't conflict with any other test case or module across forks as no two namePrefix + serviceShort combinations should ever be the same.
Feel free to object, but that would at least be my recommendation 😄

Reply:
Thanks for the recommendation, but as we see customers actualy using the hub deployment as an example for their environments I'd like to keep it with the proper WAF aligned names to avoid causing confusion.

Reply:
Fair, just be aware that this may lead to friction later down the line if patterns are ever exposed to WAF tests via PSRule (which require us to have a means to seperate test resources from actual resoures that need testing)
cc: @eriqua, @jtracey93 
Also note that because you're not using the `namePrefix` as is convention for all AVM tests you may run into blocking issues down the line for anybody running this module on their end as part of the CI and deploying resources that must be globally unique. I don't think you have any such resource right now, so that's more a heads up / context :)